### PR TITLE
PS-9681: implemented functionality to view PunchOut entry points

### PR DESCRIPTION
**Pull request by asmarovydlo**
Ticket: https://spryker.atlassian.net/browse/PS-9681
Epic: https://spryker.atlassian.net/browse/PS-8578

### CODE REVIEW

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/EntryPointsController.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Persistence\PunchoutCatalogsRepositoryInterface getRepository()
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Business\PunchoutCatalogsFacadeInterface getFacade()
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Communication\PunchoutCatalogsCommunicationFactory getFactory()
+ */
+class EntryPointsController extends AbstractController
+{
+    /**
+     * @uses \SprykerEco\Zed\PunchoutCatalogs\Communication\Controller\IndexController::indexAction()
```
Comment (vitaliykirichenkoit) [inline src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/EntryPointsController.php]: Let's keep `@uses` for cases when one constant uses the value of another, and `@see` for cases, when you want to mention the place, where this constant is used.

However, it looks really strange, that you add such annotation to protected constantly.

I assume:
    a) annotation is not needed here
or
    b) constant should be public

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/EntryPointsController.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    protected const ROUTE_PUNCHOUT_CATALOGS_CONNECTION_LIST_PAGE = '/punchout-catalogs/';
+
+    protected const PARAM_ID_PUNCHOUT_CATALOG_CONNECTION = 'id-punchout-catalog-connection';
+
+    protected const MESSAGE_CONNECTION_NOT_FOUND = 'Connection not found';
```
Comment (vitaliykirichenkoit) [inline src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/EntryPointsController.php:28]: `ERROR_MESSAGE_CONNECTION_NOT_FOUND` ?

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Table/PunchoutCatalogsConnectionsTable.php [hunk 3]
```diff
@@ -40,6 +40,11 @@ class PunchoutCatalogsConnectionsTable extends AbstractTable
      */
     protected const URL_EDIT_PUNCHOUT_CATALOG_CONNECTION = '/punchout-catalogs/index/edit';
 
+    /**
+     * @uses \SprykerEco\Zed\PunchoutCatalogs\Communication\Controller\EntryPointsController::viewAction()
+     */
+    protected const URL_PUNCHOUT_CATALOG_ENTRY_POINTS_VIEW = '/punchout-catalogs/entry-points/view';
```
Comment (vitaliykirichenkoit) [inline src/SprykerEco/Zed/PunchoutCatalogs/Communication/Table/PunchoutCatalogsConnectionsTable.php]: `URL_VIEW_PUNCHOUT_CATALOG_ENTRY_POINTS` ?

File: src/SprykerEco/Zed/PunchoutCatalogs/Presentation/EntryPoints/view.twig [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{% extends '@Gui/Layout/layout.twig' %}
```
Comment (vitaliykirichenkoit) [inline src/SprykerEco/Zed/PunchoutCatalogs/Presentation/EntryPoints/view.twig:1]: Please get approve from our FE-developer for this twig & all other FE-related things.

File: src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml [hunk 5]
```diff
@@ -24,6 +24,20 @@
         <property name="createdAt" type="string" />
     </transfer>
 
+    <transfer name="PunchoutCatalogEntryPoint">
+        <property name="store" type="Store" />
```
Comment (vitaliykirichenkoit) [inline src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml:28]: Where this field is used?

File: src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml [hunk 6]
```diff
@@ -24,6 +24,20 @@
...
+        <property name="url" type="string" />
```
Comment (vitaliykirichenkoit) [inline src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml:29]: Where this field is used?

File: src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml [hunk 7]
```diff
@@ -24,6 +24,20 @@
...
+        <property name="idCompanyBusinessUnit" type="int" />
+    </transfer>
+
+    <transfer name="Store">
+        <property name="name" type="string" />
```
Comment (vitaliykirichenkoit) [inline src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml:34]: Where this transfer and this field are used?

File: src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml (hunk 5 above)
Comment (asmarovydlo) [reply src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml:28]: It used here https://github.com/spryker-eco/punchout-catalogs/pull/9/files#diff-714fed64a6f047a764ace2851bb4a035

File: src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml (hunk 6 above)
Comment (asmarovydlo) [reply src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml:29]: It used here https://github.com/spryker-eco/punchout-catalogs/pull/9/files#diff-714fed64a6f047a764ace2851bb4a035

File: src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml (hunk 7 above)
Comment (asmarovydlo) [reply src/SprykerEco/Shared/PunchoutCatalogs/Transfer/punchout_catalog.transfer.xml:34]: It used here https://github.com/spryker-eco/punchout-catalogs/pull/9/files#diff-714fed64a6f047a764ace2851bb4a035

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/EntryPointsController.php (hunk 1 above)
Comment (asmarovydlo) [reply src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/EntryPointsController.php]: It for sure needs to be protected
Replaced by @see

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/EntryPointsController.php (hunk 2 above)
Comment (asmarovydlo) [reply src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/EntryPointsController.php:28]: Done similarly to all other stories, e.g. https://github.com/spryker-eco/punchout-catalogs/pull/7/files#diff-530ee9e2cbbd2b65737741322a482e9c

Do we have any convention for this?

Comment (vitaliykirichenkoit) [reply src/SprykerEco/Zed/PunchoutCatalogs/Communication/Controller/EntryPointsController.php:28]: If it is similar to other stories - keep it as it is. We need to keep consistency. No convention.

File: src/SprykerEco/Zed/PunchoutCatalogs/Communication/Table/PunchoutCatalogsConnectionsTable.php [hunk 8]
```diff
@@ -40,6 +40,11 @@ class PunchoutCatalogsConnectionsTable extends AbstractTable
      */
     protected const URL_EDIT_PUNCHOUT_CATALOG_CONNECTION = '/punchout-catalogs/index/edit';
 
+    /**
+     * @uses \SprykerEco\Zed\PunchoutCatalogs\Communication\Controller\EntryPointsController::viewAction()
```
Comment (vitaliykirichenkoit) [inline src/SprykerEco/Zed/PunchoutCatalogs/Communication/Table/PunchoutCatalogsConnectionsTable.php]: Please use `@see` instead.

**Review by vitaliykirichenkoit — APPROVED** [review]

**Review by odokienko — APPROVED** [review]

File: src/SprykerEco/Zed/PunchoutCatalogs/Presentation/EntryPoints/view.twig [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        {% block widget_content %}
+            <table class="table table-bordered">
+                <thead>
+                    <tr>
+                        <th>{{ 'Store Name' | trans }}</th>
+                        <th>{{ 'URL Entry Point' | trans }}</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {% for punchoutCatalogEntryPoint in punchoutCatalogEntryPoints %}
```
Comment (sergeirybalko) [inline src/SprykerEco/Zed/PunchoutCatalogs/Presentation/EntryPoints/view.twig:23]: Please add if condition before loop

Comment (asmarovydlo) [reply src/SprykerEco/Zed/PunchoutCatalogs/Presentation/EntryPoints/view.twig:23]: Discussed with @gerner-spryker, we always have at least 1 entry point

Screen Shot 2019-06-26 at 6 30 59 PM

**Review by sergeirybalko — APPROVED** [review]

**Review by demkos — APPROVED** [review]

**Pull request merged**